Tests/SwiftAlgorithmsTests/IntersperseTests.swift

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import Algorithms
+
+final class IntersperseTests: XCTestCase {
+  func testString() {
+    let interspersed = "ABCDE".interspersed(with: "-")
+    XCTAssertEqualSequences(interspersed, "A-B-C-D-E")
+    validateIndexTraversals(interspersed)
+  }
+
+  func testStringEmpty() {
+    let interspersed = "".interspersed(with: "-")
+    XCTAssertEqualSequences(interspersed, "")
+    validateIndexTraversals(interspersed)
+  }
```

Comment:
Could you add a test that validates against empty and non-empty sequences? `(0...).prefix(_)` is an easy way to create something that will call through to sequence-based iteration.

Reply:
Ah that makes sense, so that it doesn't go through any of the Collection code? If I've understood correctly, this is in 
d4bdbc4. 🙂